File: LeetCodeTasks/3006_find-beautiful-indices-in-the-given-array-i/solutions/java/solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> beautifulIndices(String s, String a, String b, int k) {
        int n = s.length();
        int lenA = a.length();
        int lenB = b.length();

        // Lists to hold indices where substring a and b are found
        List<Integer> indicesA = new ArrayList<>();
        List<Integer> indicesB = new ArrayList<>();

        // Find all occurrences of a
        for (int i = 0; i <= n - lenA; i++) {
            if (s.substring(i, i + lenA).equals(a)) {
                indicesA.add(i);
            }
        }

        // Find all occurrences of b
        for (int i = 0; i <= n - lenB; i++) {
            if (s.substring(i, i + lenB).equals(b)) {
                indicesB.add(i);
            }
        }

        List<Integer> result = new ArrayList<>();
        
        // Iterate over each index where 'a' is found and check proximity to any 'b'
        for (int idxA : indicesA) {
            // Check if within k distance of any index of b
            boolean found = false;
            for (int idxB : indicesB) {
                if (Math.abs(idxB - idxA) <= k) {
                    found = true;
                    break;
                }
            }
            if (found) {
                result.add(idxA);
            }
        }

        return result;
    }
}

File: LeetCodeTasks/3006_find-beautiful-indices-in-the-given-array-i/solutions/cpp/solution.cpp
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<int> beautifulIndices(string s, string a, string b, int k) {
        vector<int> indicesA, indicesB, result;
        int lenA = a.size(), lenB = b.size();
        
        // Collect all occurrences of a in s
        for (int i = 0; i <= int(s.length()) - lenA; ++i) {
            if (s.substr(i, lenA) == a) {
                indicesA.push_back(i);
            }
        }
        
        // Collect all occurrences of b in s
        for (int j = 0; j <= int(s.length()) - lenB; ++j) {
            if (s.substr(j, lenB) == b) {
                indicesB.push_back(j);
            }
        }
        
        // For each index in indicesA, check for any valid index in indicesB
        for (int i : indicesA) {
            bool found = false;
            for (int j : indicesB) {
                if (abs(i - j) <= k) {
                    found = true;
                    break;
                }
            }
            if (found) {
                result.push_back(i);
            }
        }
        
        return result;
    }
};
